spec/migrations/20241017013023_reencrypt_password_scramsha_spec.rb

```diff
@@ -9,9 +9,9 @@
 
       username = ActiveRecord::Base.connection_db_config.configuration_hash[:username]
 
-      expect(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).exactly(10).times.and_call_original
-      expect(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).and_wrap_original do |original_method, *args, &block|
-        original_method.call(*args, &block).dup.tap { |i| i[:password] ||= "abc" }
+      allow(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).and_wrap_original do |original_method, *args, &block|
+        # set a value for any calls originating from the migration file, not from rails itself
+        original_method.call(*args, &block).dup.tap {|i| i[:password] ||= "abc" if caller_locations.any? {|loc| loc.path.include?("db/migrate/20241017013023_reencrypt_password_scramsha.rb")} }
```

Comment:
Extremely minor but would it be worth extracting this helper code into a new helper method and then using that here?

https://github.com/ManageIQ/manageiq-schema/blob/7af242029e4c111510c37bf32911e3749414a51c/spec/support/migration_name_helper.rb#L2-L3

Reply:
sure, I'll take a look when I come back to this.  👍